tensorflow-tools/src/main/java/org/tensorflow/tools/Shape.java

```diff
@@ -97,6 +178,17 @@ public int hashCode() {
     return dimensionSizes != null ? Arrays.hashCode(dimensionSizes) : super.hashCode();
   }
 
+  /**
+   * Equals implementation for shapes. Two shapes are considered equal iff:
+   * <p><ul>
+   * <li>the number of dimensions is defined and equal for both
+   * <li>the size of each dimension is defined and eqaul for both
+   * </ul><p>
+   * If either shape has an unknown dimension (even if it is the same in both) or if either
+   * shape has an unknown number of dimensions (even if both return {@code true} for
+   * {@link Shape#isUnknown()}), they are not considered equal. However, a shape will always
+   * equal itself, event it is unknown or contains unknown dimensions.
```

Comment:
`event it is...` you meant `even if it is`?

Reply:
Yup, that's what I meant. Fixed it.